src/Runner/Runner.php

```diff
@@ -47,68 +63,216 @@ final class Runner
 
     private CacheManagerInterface $cacheManager;
 
-    private bool $isDryRun;
-
     private LinterInterface $linter;
 
     /**
-     * @var \Traversable<\SplFileInfo>
+     * @var ?iterable<\SplFileInfo>
      */
-    private $finder;
+    private $fileIterator;
+
+    private int $fileCount;
 
     /**
      * @var list<FixerInterface>
      */
     private array $fixers;
 
-    private bool $stopOnViolation;
-
     /**
-     * @param \Traversable<\SplFileInfo> $finder
-     * @param list<FixerInterface>       $fixers
+     * @param null|iterable<\SplFileInfo> $fileIterator
+     * @param list<FixerInterface>        $fixers
      */
     public function __construct(
-        \Traversable $finder,
+        RunnerConfig $runnerConfig,
+        ?iterable $fileIterator,
         array $fixers,
         DifferInterface $differ,
         ?EventDispatcherInterface $eventDispatcher,
         ErrorsManager $errorsManager,
         LinterInterface $linter,
-        bool $isDryRun,
         CacheManagerInterface $cacheManager,
-        ?DirectoryInterface $directory = null,
-        bool $stopOnViolation = false
```

Comment:
question: what is the pattern determining which property went to `RunnerConfig` and which did not?